src/main/java/Menu.java

```diff
@@ -17,27 +19,58 @@ public class Menu {
     private static int numberOfStudents;
 
     public static void main(String[] args) {
-        System.out.println(FigletFont.convertOneLine(getValidNameFor("school")));
+        String name = getValidNameFor("school");
+        System.out.println(FigletFont.convertOneLine(name));
+        School school = new School(name);
         //System.out.println("Option Selected: " + showPrincipalMenuAndRetrieveOption(scanner));
         numberOfCourses = getNumberOfEntity("courses", MaxValue.MAX_COURSES_TO_CREATE.getValue());
         numberOfTeachers = getNumberOfEntity("teachers", MaxValue.MAX_TEACHER_TO_CREATE.getValue());
         numberOfStudents = getNumberOfEntity("students", MaxValue.MAX_STUDENT_TO_CREATE.getValue());
+        registerStudents(school);
+    }
+
+    private static void registerStudents(School school) {
```

Comment:
If we use variables in a loop, these should go out of the loop, regarding garbage collector! In the loop just reassign but not create new variables. as ...
String name, address, email; 

But better approach would be just call the methods in the constructor's call  as..
Student student = new Student(getValidNameFor("student"), getValidAddress(), getValidEmail());

Reply:
I believe it was more legible before but I don't have a strong opinion...

Reply:
OK, but at least considerer define the variables outside of every the loop. In java is how it should be regarding how garbage collector works. We could talk with the teacher to discuss about it I think it is very interesting topic!

Reply:
This is partially correct and depends on the context :) I read that with String, reassignment typically involves creating new objects anyway, because Strings are immutable.